src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -144,7 +147,13 @@ def load_command_table(self, command_loader):
                 _parser=command_parser)
 
     def validation_error(self, message):
-        az_error = ValidationError(message)
+        # Get command to provide to error help
+        full_command = self.prog + ' ' + self.full_raw_command
+
+        cli_ctx = self.cli_ctx or (self.cli_help.cli_ctx if self.cli_help else None)
```

Comment:
Why do we need add this logic?

Reply:
This is copied from other function. I refactor the code and hope that is clearer